content/blog/2021/08/19/WritingFastConstraintStreamsSecretRecipe.adoc

```diff
@@ -0,0 +1,119 @@
+= Constraint Streams get some more love
+triceo
+2021-08-19
+:page-interpolate: true
+:jbake-type: post
+:jbake-tags: constraint, feature
+
+I find it hard to believe that it's been over two years already since we've introduced https://docs.optaplanner.org/latestFinal/optaplanner-docs/html_single/index.html#constraintStreams[Constraint Streams] into OptaPlanner; how time flies!
+Everyone's favorite API is constantly being improved, and only recently have we merged a whole lot of improvements to make it even easier to use.
+Let's take a closer look at some of them, available as of OptaPlanner 8.11.0.Final, coming soon to a mirror near you.
```

Comment:
The first paragraph should entice me to read further. In 2-3 sentences it should convince me that I am going to learn useful things by reading further. Basically ROI focused. An high-level ROI with low level proofs works well typically. I think the ROI here is "faster", "more powerful" (proof: "5 consecutive shifts in a row") etc.

The "how time flies" bit makes it human. That's great for the start of the 2th paragraph, after a strong opening, showing a softer, more human side.

The available as of OptaPlanner 8.11 part is for in the conclusion, I'd argue. Good new first: First you sell them on the thing you're selling, then you tell on the limitations of it (but never forget to tell the bad news too, of course).